### PR TITLE
Animated SymbolLayer icon movement example (pin drop)

### PR DESCRIPTION
**Pull request by langsmith**
This pr adds an example of combining SymbolLayer icons with the Android system's ValueAnimator and interpolator animation for a fun pin drop effect. The interpolator movement can also be used with other types of map layers, such as a LineLayer or CircleLayer.

ezgif com-resize 1

resolves https://github.com/mapbox/mapbox-android-demo/issues/880  by showing how `iconTranslate` can be used to adjust icon location.

### CODE REVIEW

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/ValueAnimatorIconAnimationActivity.java [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+  private ValueAnimator animator;
+  private boolean firstRunThrough;
+  private boolean animationHasStarted;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    currentSelectedTimeInterpolator = new BounceInterpolator();
+    firstRunThrough = true;
```
Comment (tobrun) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/ValueAnimatorIconAnimationActivity.java]: can be assigned inline in class member definition

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/ValueAnimatorIconAnimationActivity.java [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+  private TimeInterpolator currentSelectedTimeInterpolator;
+  private ValueAnimator animator;
+  private boolean firstRunThrough;
+  private boolean animationHasStarted;
+
+  @Override
+  protected void onCreate(Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+
+    currentSelectedTimeInterpolator = new BounceInterpolator();
```
Comment (tobrun) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/ValueAnimatorIconAnimationActivity.java]: can be assigned inline in class member definition

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/ValueAnimatorIconAnimationActivity.java [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+  }
+
+  @Override
+  public void onMapReady(@NonNull final MapboxMap mapboxMap) {
+    mapboxMap.setStyle(Style.LIGHT, new Style.OnStyleLoaded() {
+      @Override
+      public void onStyleLoaded(@NonNull Style style) {
+        ValueAnimatorIconAnimationActivity.this.style = style;
+        mapView.addOnDidFinishRenderingMapListener(ValueAnimatorIconAnimationActivity.this);
+        initLayerIcon(style);
```
Comment (tobrun) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/ValueAnimatorIconAnimationActivity.java]: nit can be added with the StyleBuilder#withIcon

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/ValueAnimatorIconAnimationActivity.java [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+        initDataSource(style);
```
Comment (tobrun) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/ValueAnimatorIconAnimationActivity.java]: nit can be added with the StyleBuilder#withSource/Layer

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/ValueAnimatorIconAnimationActivity.java [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+    );
+  }
+
+  /**
+   * Initialize and start the animation.
+   *
+   * @param desiredTimeInterpolator the type of Android system movement to animate the
+   *                                SymbolLayer icons with.
+   */
+  private void initAnimation(TimeInterpolator desiredTimeInterpolator) {
```
Comment (tobrun) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/ValueAnimatorIconAnimationActivity.java]: need to check if animator is not null, if yes call cancel on it

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/ValueAnimatorIconAnimationActivity.java [hunk 6]
```diff
@@ -0,0 +1,290 @@
...
+        firstRunThrough = false;
+        resetIcons();
+      }
+    });
+
+    FloatingActionButton decelerateInterpolatorFab = findViewById(R.id.fab_decelerate_interpolator);
+    decelerateInterpolatorFab.setOnClickListener(new View.OnClickListener() {
+      @Override
+      public void onClick(View view) {
+
```
Comment (tobrun) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/ValueAnimatorIconAnimationActivity.java]: nit whiteline

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/ValueAnimatorIconAnimationActivity.java [hunk 7]
```diff
@@ -0,0 +1,290 @@
...
+        firstRunThrough = false;
+        resetIcons();
+      }
+    });
+  }
+
+  private void resetIcons() {
+    if (!firstRunThrough) {
+      animationHasStarted = false;
+      style.removeLayer("symbol-layer-id");
```
Comment (tobrun) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/ValueAnimatorIconAnimationActivity.java]: can be put in a constant

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/ValueAnimatorIconAnimationActivity.java (hunk 5 above)
Comment (langsmith) [reply MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/ValueAnimatorIconAnimationActivity.java]: Fixed

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/ValueAnimatorIconAnimationActivity.java (hunk 6 above)
Comment (langsmith) [reply MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/ValueAnimatorIconAnimationActivity.java]: Removed

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/ValueAnimatorIconAnimationActivity.java (hunk 7 above)
Comment (langsmith) [reply MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/ValueAnimatorIconAnimationActivity.java]: Fixed

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/ValueAnimatorIconAnimationActivity.java (hunk 3 above)
Comment (langsmith) [reply MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/ValueAnimatorIconAnimationActivity.java]: Fixed

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/ValueAnimatorIconAnimationActivity.java (hunk 4 above)
Comment (langsmith) [reply MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/ValueAnimatorIconAnimationActivity.java]: Added source via StyleBuilder, but keeping `initSymbolLayer()` as is if it's ok. The example is so fine-grained with FABS, etc. that it's easier to initialize the layer a bit after the style is actually created..

**Comment by langsmith** [comment]
Ok, feedback has been addressed @tobrun

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/ValueAnimatorIconAnimationActivity.java (hunk 1 above)
Comment (langsmith) [reply MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/ValueAnimatorIconAnimationActivity.java]: Fixed

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/ValueAnimatorIconAnimationActivity.java (hunk 2 above)
Comment (langsmith) [reply MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/ValueAnimatorIconAnimationActivity.java]: Fixed

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/ValueAnimatorIconAnimationActivity.java (hunk 4 above)
Comment (langsmith) [reply MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/labs/ValueAnimatorIconAnimationActivity.java]: Screen Shot 2019-03-29 at 2 16 46 PM
Screen Shot 2019-03-29 at 2 16 42 PM

**Pull request merged**